### PR TITLE
Port Cppcheck from ERT to Buttercup

### PR DESCRIPTION
**Pull request by Simplify**
I changed last two tests, so please review. This is last language parser from ERT.

### CODE REVIEW

File: test/specs/languages/test-cpp.el [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                  :level 'warning
+                  :message "The expression \"x\" is of type 'bool' and it is compared against a integer value that is neither 1 nor 0."
+                  :id "comparisonOfBoolWithInt"))))
+
+      (it "parses Cppcheck XML without errors using automatic parser selection"
+        (expect (flycheck-parse-cppcheck cppcheck-xml-without-errors
+                                         'checker 'buffer)
+                :to-be-equal-flycheck-errors ()))
+
+      (it "parses Cppcheck XML without errors using buildin parser"
```
Comment (lunaryorn) [inline test/specs/languages/test-cpp.el]: `buildin` -> `builtin`

File: test/specs/languages/test-cpp.el [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                  :line 2
+                  :column nil
+                  :level 'warning
+                  :message "The expression \"x\" is of type 'bool' and it is compared against a integer value that is neither 1 nor 0."
+                  :id "comparisonOfBoolWithInt"))))
+
+      (it "parses Cppcheck XML without errors using automatic parser selection"
+        (expect (flycheck-parse-cppcheck cppcheck-xml-without-errors
+                                         'checker 'buffer)
+                :to-be-equal-flycheck-errors ()))
```
Comment (lunaryorn) [inline test/specs/languages/test-cpp.el]: `()` -> `nil`.

File: test/specs/languages/test-cpp.el [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+      (it "parses Cppcheck XML without errors using buildin parser"
+        (let ((flycheck-xml-parser #'flycheck-parse-xml-region))
+          (expect (flycheck-parse-cppcheck cppcheck-xml-without-errors
+                                           'checker 'buffer)
+                  :to-be-equal-flycheck-errors ()))))))
```
Comment (lunaryorn) [inline test/specs/languages/test-cpp.el]: `()` -> `nil`

File: test/specs/languages/test-cpp.el [hunk 4]
```diff
@@ -0,0 +1,106 @@
+;;; test-cpp.el --- Flycheck Specs: C++      -*- lexical-binding: t; -*-
```
Comment (lunaryorn) [inline test/specs/languages/test-cpp.el]: We use `c++` to refer to this language all throughout our tests.  Please rename this file to `test-c_c++.el` if possible.

**Comment by lunaryorn** [comment]
@Simplify Awesome.  Just a couple of minor issues :blush:

Many thanks for your work on our tests!

**Comment by Simplify** [comment]
@lunaryorn Done.

**Comment by lunaryorn** [comment]
@Simplify Great, merged!

**Comment by Simplify** [comment]
Btw, I usually avoid `+` in filenames, because of Windows support, but it turns out that `+` is restriction only on FAT, not on NTFS. FAT is no longer used on new Windows systems, except for memory cards and USB sticks, so I guess that is no longer a problem.

**Comment by lunaryorn** [comment]
@Simplify Flycheck doesn't support Windows :wink:

**Pull request merged**